wandb/util.py

```diff
@@ -1848,10 +1850,42 @@ def get_core_path() -> str:
     return core_path
 
 
+def working_set() -> Iterable:
+    """Return the working set of installed distributions.
+
+    This function returns a list of namedtuples representing the installed distributions.
+    Each namedtuple has two fields: `key` and `version`. If the `importlib.metadata` module
+    is available, it uses that to get the working set. Otherwise, it falls back to the
+    `importlib_metadata` module.
+    """
+    try:
+        from importlib.metadata import distributions
+    except ImportError:  # Python 3.7
+        from importlib_metadata import distributions  # type: ignore
+    import collections
+
+    WorkingSet = collections.namedtuple("WorkingSet", "key version")
+    for d in distributions():
+        yield WorkingSet(key=d.metadata["Name"], version=d.version)
+
+
+def parse_version(version: str) -> "packaging.version.Version":
+    """Parse a version string into a version object.
+
+    This function is a wrapper around the `packaging.version.parse` function, which
+    is used to parse version strings into version objects. If the `packaging` library
+    is not installed, it falls back to the `pkg_resources` library.
+    """
+    try:
+        from packaging.version import parse as parse_version  # type: ignore
+    except ImportError:
+        from pkg_resources import parse_version
```

Comment:
Is this for py37 support? It might be useful to say that down explicitly in the docstring, to make it clearer that this wrapper can be removed after we drop py37 support.

Reply:
no, packaging is supported in py37. It's just not part of the standard library packages in python, and didn't want to add it as a dependency (but we could, as eventually `pkg_resources` would be probably removed). Currently, i just relay on the user's env, if they have packaging installed use it if not fallback to pkg_resources i can  add comment around it